src/inference/program_models/Topyfic/Topyfic.py

```diff
@@ -0,0 +1,228 @@
+        work_dir: str
+            path to directory to store outputs.
+        prog_key: 
+            index for the anndata object (mdata[prog_key]) in the mudata object.
+        data_key: str
+            index of the genomic data anndata object (mdata[data_key]) in the mudata object.
+        layer: str (default: X)
+            anndata layer (mdata[data_key].layers[layer]) where the data is stored.
+        config_path: str
+            path to gin configurable config file containing method specific parameters.
+        inplace: Bool (default: True)
+            update the mudata object inplace or return a copy
+
+    RETURNS:
+        if not inplace:
+            mdata
+    
+    """
+
+    # Load method specific parameters
+    try:
+        gin.parse_config_file(config_path)
+    except:
+        raise ValueError('gin config file could not be found')
+
+    if not inplace:
+        mdata = mudata.MuData({data_key: mdata[data_key].copy()})
+
+    # Create output directory for cNMF results
+    if work_dir is not None:
+        try:
+            os.makedirs(work_dir, exist_ok=True)
+        except:
+            raise ValueError('Work directory location does not exist.')
+
+    # Compute (structure this as neccessary)
+    # Not that method specific params loaded via gin do not
+    # have to be passed explicity
+    train, top_model, analysis_top_model = \
+        run_Topyfic_(mdata[data_key],
+                     layer=layer,
+                     output_dir=work_dir)
+
+    cell_program_scores = top_model.get_gene_weights().T
+    cell_topic_participation = analysis_top_model.cell_participation
+    cell_topic_participation.var = pd.concat([cell_topic_participation.var, cell_program_scores], axis=1)
+
+    # Store cell x program and program x feature matrices
+    mdata.mod[prog_key] = cell_topic_participation
+
+    if not inplace:
+        return mdata
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('mudataObj_path')
+    parser.add_argument('--work_dir', default='./', type=str)
+    parser.add_argument('-pk', '--prog_key', default='program_inference_method', typ=str)
+    parser.add_argument('-dk', '--data_key', default='rna', typ=str)  # could be atac
+    parser.add_argument('--layer', default='X', type=str)  # layer of data anndata to use for inference
+    parser.add_argument('--config_path', default='./Topyfic_config.gin', type=str)
+    parser.add_argument('--output', action='store_false')
+
+    args = parser.parse_args()
+
+    mdata = mudata.read(args.mudataObj_path)
+    run_program_inference_method(mdata,
+                                 work_dir=args.work_dir,
+                                 prog_key=args.prog_key,
+                                 data_key=args.data_key,
+                                 layer=args.layer,
+                                 config_path=args.config_path,
+                                 inplace=args.output)
```

Comment:
Could you remove the parts of the template that are not being used and also modify this section to call the Topyfic function instead of the template.